src/Sulu/Bundle/MediaBundle/Resources/public/js/components/area-selection/main.js

```diff
@@ -0,0 +1,489 @@
+         * gets used to position the area. Otherwise the area gets placed into the center
+         * of the image.
+         */
+        computeInitialAreaCoordinates: function() {
+            var coordinates;
+
+            if (!!this.data) {
+                coordinates = this.dataToCoordinates(this.data);
+            } else {
+                coordinates = this.dataToCoordinates(this.getMaximumCenteredData());
+            }
+
+            this.area.coordinates = coordinates;
+            this.setAreaPosition(coordinates);
+            this.setAreaSize(coordinates);
+        },
+
+        /**
+         * Sets the position of the area. Moreover the logical position coordinates are calculated from the passed size.
+         * The passed size object gets validated (and probably changed through constraints) before applying the position.
+         *
+         * @param position {Object} an object with 'x' and 'y' property
+         */
+        setAreaPosition: function(position) {
+            position = this.sandbox.util.extend(true, {}, this.area.coordinates, position);
+            position = this.getConstrainedPosition(position);
+
+            // Set the logical data (calculated from the coordinates)
+            this.$el.data('area', _.extend(this.$el.data('area'), {
+                x: Math.floor(position.x * this.originalWidth / this.$frame.width()),
+                y: Math.floor(position.y * this.originalHeight / this.$frame.height())
+            }));
+
+            // Set the physical coordinates of the area
+            this.area.$el.css('left', Math.round(position.x) + 'px');
+            this.area.$el.css('top', Math.round(position.y) + 'px');
+            this.area.coordinates.x = position.x;
+            this.area.coordinates.y = position.y;
+
+            // Give the backdrop the same coordinates as the area
+            this.$backdrop.css('left', Math.round(position.x) + 'px');
+            this.$backdrop.css('top', Math.round(position.y) + 'px');
+        },
+
+        /**
+         * Sets the size of the area. Moreover the logical size coordinates are calculated from the passed size.
+         * The passed size object gets validated (and probably changed) before applying the size.
+         *
+         * @param size {Object} an object with 'width' and 'height' property
+         */
+        setAreaSize: function(size) {
+            size = this.sandbox.util.extend(true, {}, this.area.coordinates, size);
+            size = this.getConstrainedSize(size);
+
+            // Set the logical data (calculated from the size)
+            this.$el.data('area', _.extend(this.$el.data('area'), {
+                width: Math.floor(size.width * this.originalWidth / this.$frame.width()),
+                height: Math.floor(size.height * this.originalHeight / this.$frame.height())
+            }));
+
+            if ((!!this.guideWidth && this.$el.data('area').width <= this.guideWidth) ||
```

Comment:
I prefer to have the logical combinators in the front of the next line instead of the end, since they are important and easily overseen when placed at the end of a line.